src/__tests__/fsm.tests.ts

```diff
@@ -0,0 +1,91 @@
+import fsm from '../lib/fsm';
+
+describe('fsm-tests', () => {
+  it('should export fsm', () => {
+    let sut = new fsm('initial', []);
+
+    expect(sut).not.toBe(undefined);
+    expect(sut).not.toBe(null);
+  });
+});
+
+describe('fsm-tests', () => {
```

Comment:
duplicated:

You can do:

```
describe('DeterministicStateMachine', () => {

  describe('packaging', () => {
     it('should do A', () => {
        // TEST CASE A for packaging
     });
     it('should do B', () => {
        // TEST CASE B for packaging
     });
  });

  it('should do C', () => {
    // TEST CASE C for global
  });
});
```
which results in:
DeterministicStateMachine packaging should do A
DeterministicStateMachine packaging should do B
DeterministicStateMachine should do C